### PR TITLE
Track canal replication delay with a gauge metric

### PR DESCRIPTION
**Pull request by bejelith**
Need a way to track replication delay without duplicating code over each hook

### CODE REVIEW

File: canal/canal.go [hunk 1]
```diff
@@ -166,6 +168,10 @@ func (c *Canal) prepareDumper() error {
 	return nil
 }
 
+func (c *Canal) GetDelay() uint32 {
+	return c.delay
```
Comment (siddontang) [inline canal/canal.go]: may meet data race?

File: canal/sync.go [hunk 2]
```diff
@@ -175,6 +179,10 @@ func (c *Canal) runSyncBinlog() error {
 	return nil
 }
 
+func (c *Canal) updateReplicationDelay(ev *replication.BinlogEvent) {
+	c.delay = uint32(time.Now().Unix()) - ev.Header.Timestamp
```
Comment (siddontang) [inline canal/sync.go]: maybe we should use atomic to save delay

Comment (bejelith) [reply canal/sync.go]: 👍🏻 Yeah somehow didnt think about it

**Comment by siddontang** [comment]
CI failed @bejelith

**Comment by bejelith** [comment]
@siddontang PTAL

**Pull request merged**